### PR TITLE
Feature/publish block

### PR DESCRIPTION
**Pull request by aratz-lasa**
# What 🕵️‍♀️
This PR publishes the signed beacon block after a payload delivery

# Why 🔑
This reduces chances of missing payloads due to high-latencies between relay and mev-boost

# Testing 🧪
- [ ] `go test -race ./...` from root
- [x] test on testnet

### CODE REVIEW

**Review by dmarzzz — APPROVED** [review]
LGTM! Can we include a reference to the flashbots relay PR in this PR description to give credit for the idea?

File: pkg/relay/relay.go [hunk 1]
```diff
@@ -287,6 +296,15 @@ func (rs *Relay) GetPayload(ctx context.Context, payloadRequest *types.SignedBli
 
 	// defer put delivered datastore write
 	go func(rs *Relay, slot structs.Slot, trace structs.DeliveredTrace) {
+		if rs.config.PublishBlock {
```
Comment (lukanus) [inline pkg/relay/relay.go:299]: I'm not super sure about that, but if we're doing that - I think this should happen after rs.d.putDelivered - not before

File: pkg/relay/relay.go [hunk 2]
```diff
@@ -287,6 +296,15 @@ func (rs *Relay) GetPayload(ctx context.Context, payloadRequest *types.SignedBli
...
+			beaconBlock := structs.SignedBlindedBeaconBlockToBeaconBlock(payloadRequest, payload.Payload.Data)
+			if err := rs.beacon.PublishBlock(beaconBlock); err != nil {
+				logger.WithError(err).Warn("fail to publish block to beacon node")
+			} else {
+				logger.Info("published block to beacon node")
```
Comment (lukanus) [inline pkg/relay/relay.go:304]: also `rs.l.` :)

File: pkg/relay/relay.go (hunk 1 above)
Comment (aratz-lasa) [reply pkg/relay/relay.go:299]: Why after `PutDelivered`? PutDelivered is only for updating the Data API

File: pkg/relay/relay.go (hunk 2 above)
Comment (aratz-lasa) [reply pkg/relay/relay.go:304]: Why? `logger` is initialized in the beginning of the method, and it contains fields for describing the request

File: pkg/relay/relay.go (hunk 1 above)
Comment (lukanus) [reply pkg/relay/relay.go:299]: Because the integrity of dataAPI imho trumps block publishing, but if you like to keep it like that, not a big deal

Comment (aratz-lasa) [reply pkg/relay/relay.go:299]: And does not affect to whether the call will be successful or not. In case of error we simply log it

Comment (aratz-lasa) [reply pkg/relay/relay.go:299]: So, you suggest not to publish the block it if the `PutDelivered` call fails? I think the Data API should be relatively decoupled from the relay operations.

Comment (lukanus) [reply pkg/relay/relay.go:299]: I was thinking about something more like a precaution, like if publish block (randomly?) panics, to be paranoic about data api having it published.

Comment (aratz-lasa) [reply pkg/relay/relay.go:299]: Okay sure, paranoid-drive-development is always good ;) It does not have any drawbacks, so I'll move it to after the call

File: pkg/relay/relay.go (hunk 2 above)
Comment (lukanus) [reply pkg/relay/relay.go:308]: 👍

**Pull request merged**